server.js

```diff
@@ -4,162 +4,243 @@
-  let gasProperty = new Property(thing,
-                                'gas',
-                                new Value({}, () => {}),
-                                {type: 'object'});
+
+  const gasProperty = new Property(
+    thing,
+    'gas',
+    new Value({}),
+    {
+      type: 'object',
+    });
   thing.addProperty(gasProperty);
 
-  let luminosityProperty = new Property(thing,
-                                        'luminosity',
-                                        new Value(0, () => {}),
-                                        {type: 'number',
-                                         unit: 'lux'});
+  const luminosityProperty = new Property(
+    thing,
+    'luminosity',
+    new Value(0),
+    {
+      type: 'number',
+      unit: 'lux',
+    });
   thing.addProperty(luminosityProperty);
 
-  let batteryLevelProperty = new Property(thing,
-                                          'battery',
-                                          new Value(0, () => {}),
-                                          {type: 'number',
-                                           unit: 'percent'});
+  const batteryLevelProperty = new Property(
+    thing,
+    'battery',
+    new Value(0),
+    {
+      type: 'number',
+      unit: 'percent',
+    });
   thing.addProperty(batteryLevelProperty);
 
-  let colorProperty = new Property(thing, 'color', new Value('#00f', (value) => {
-    let color = Color(value);
-    this.thingy.led_set(color.object(), function(error) {
-      console.log('LED color changed ' + ((error) ? error : ''));
+  const colorProperty = new Property(
+    thing,
+    'color',
+    new Value('#0000FF'),
+    {
+      type: 'string',
     });
-  }), {type: 'string'});
   thing.addProperty(colorProperty);
 
-  console.log('setup');
-
-  thingy.connectAndSetUp(function(error) {
-    console.log('Connected! ' + error ? error : '');
-
-    thingy.on('temperatureNotif', temperatureProperty.setValue.bind(temperatureProperty));
-    thingy.on('pressureNotif', pressureProperty.setValue.bind(pressureProperty));
-    thingy.on('humidityNotif', humidityProperty.setValue.bind(humidityProperty));
-    thingy.on('gasNotif', gasProperty.setValue.bind(gasProperty));
-    thingy.on('colorNotif', value => {
-      let color = Color({r: value.red,
-                         g: value.green,
-                         b: value.blue});
-      colorProperty.setValue(value.hex());
-      luminosityProperty.setValue(value['clear']);
+  const ledColorProperty = new Property(
+    thing,
+    'ledColor',
+    new Value('#0000FF', (value) => {
+      const color = Color(value);
+      thingy.led_set(color.object(), (error) => {
+        if (error) {
+          console.log('Failed to set LED color:', error);
+        }
+      });
+    }),
+    {
+      type: 'string',
     });
+  thing.addProperty(ledColorProperty);
 
-    thingy.on('batteryLevelChange', batteryLevelProperty.setValue.bind(batteryLevelProperty));
+  thingy.connectAndSetUp((error) => {
```

Comment:
I'd like to keep the "Connected!" log here since there's a 50% chance that the thingy:52 fails to connect properly and never gets from discovered to connected.